src/BDCS/API/V0.hs

```diff
@@ -2029,6 +2034,90 @@ composeStatus ServerConfig{..} uuids =
 
 
+data ComposeInfoResponse = ComposeInfoResponse {
+    cirCommit      :: T.Text,                                           -- ^ Blueprint git commit hash
+    cirBlueprint   :: Recipe,                                           -- ^ Frozen Blueprint
+    cirType        :: T.Text,                                           -- ^ Build type (tar, etc.)
+    cirBuildId     :: T.Text,                                           -- ^ Build UUID
+    cirQueueStatus :: T.Text                                            -- ^ Build queue status
+} deriving (Show, Eq)
+
+instance ToJSON ComposeInfoResponse where
+  toJSON ComposeInfoResponse{..} = object
+    [ "commit"       .= cirCommit
+    , "blueprint"    .= cirBlueprint
+    , "compose_type" .= cirType
+    , "id"           .= cirBuildId
+    , "queue_status" .= cirQueueStatus
+    ]
+
+instance FromJSON ComposeInfoResponse where
+  parseJSON = withObject "/compose/info response" $ \o -> do
+    cirCommit      <- o .: "commit"
+    cirBlueprint   <- o .: "blueprint"
+    cirType        <- o .: "compose_type"
+    cirBuildId     <- o .: "id"
+    cirQueueStatus <- o .: "queue_status"
+    return ComposeInfoResponse{..}
+
+-- | /api/v0/compose/info/<uuid>
+--
+-- Get detailed information about the compose. The returned JSON string will
+-- contain the following information:
+--
+--   * id - The uuid of the comoposition
+--   * config - containing the configuration settings used to run Anaconda
+--   * blueprint - The depsolved blueprint used to generate the kickstart
+--   * commit - The (local) git commit hash for the blueprint used
+--   * deps - The NEVRA of all of the dependencies used in the composition
+--   * compose_type - The type of output generated (tar, iso, etc.)
+--   * queue_status - The final status of the composition (FINISHED or FAILED)
+--
+-- Example::
+--
+-- > {
+-- >   "commit": "7078e521a54b12eae31c3fd028680da7a0815a4d",
+-- >   "compose_type": "tar",
+-- >   "id": "c30b7d80-523b-4a23-ad52-61b799739ce8",
+-- >   "queue_status": "FINISHED",
+-- >   "blueprint": {
+-- >     "description": "An example kubernetes master",
+-- >     ...
+-- >   }
+-- > }
+--
+composeInfo :: ServerConfig -> String -> Handler ComposeInfoResponse
+composeInfo ServerConfig{..} uuid = do
+    result <- liftIO $ runExceptT $ mkComposeStatus cfgResultsDir (cs uuid)
+    case result of
+        Left _                  -> throwError invalid_uuid
+        Right ComposeStatus{..} -> do
+            config <- liftIO $ readComposeConfigFile results_dir
+            case config of
+                Left _                  -> throwError config_error
+                Right ComposeConfig{..} -> do
+                    frozen <- liftIO $ readFrozenBlueprintFile results_dir
+                    case frozen of
+                        Left _       -> throwError frozen_error
+                        Right recipe -> return $ ComposeInfoResponse ccCommit recipe ccExportType (cs uuid) (queueStatusText csQueueStatus)
```

Comment:
Likewise on being stairsteppy here, but I think we can probably play around with this later.